overlord/hookstate/ctlcmd/set.go

```diff
@@ -235,13 +235,15 @@ func setRegistryValues(ctx *hookstate.Context, plugName string, requests map[str
 		return fmt.Errorf("cannot set registry: %v", err)
 	}
 
-	tx, err := registrystate.RegistryTransaction(ctx, view.Registry())
+	if registrystate.IsRegistryHook(ctx) && !strings.HasPrefix(ctx.HookName(), "change-view-") {
+		return fmt.Errorf("cannot modify registry in %q hook", ctx.HookName())
+	}
+
+	regCtx := registrystate.NewContext(ctx)
```

Comment:
we cannot use onDone on regCtx here, so not sure why we don't pass ctx and GetTransaction makes then its Context?

Reply:
The registrystate.Context is a way to defer the creation of the change or updating of state until the right time. For a registry hook, that might be when it ends. The API will not block but it will trigger it and then get the change (it'll need minor changes for that) and include it in the http response